pytorch_lightning/plugins/base_plugin.py

```diff
@@ -19,7 +19,14 @@
 
 
 class Plugin(ABC):
-    """Basic class for all precision- and training type plugins."""
+    """
+    Base class for all precision and training type plugins.
+
+    .. deprecated::v1.5
+        Will be removed in v1.7.0.
+        Call TrainingTypePlugin or PrecisionPlugin directlly
+
+    """
```

Comment:
i think we should remove the base plugin directly in this PR. i don't think it's worth doing a deprecation cycle if it's not referenced anywhere else in the trainer

Reply:
Sounds good!